code/modules/cargo/exports.dm

```diff
@@ -43,11 +47,18 @@ Credit dupes that require a lot of manual work shouldn't be removed, unless they
 		var/sold = FALSE
 		if(QDELETED(thing))
 			continue
+
+		//But first I'm going to check if you're getting the full payout.
+		if(thing.GetComponent(/datum/component/pricetag))
+			profit_ratio = (thing.GetComponent(/datum/component/pricetag)).profit_ratio
```

Comment:
You shouldn't be using GetComponent to do things that are part of the functionality needed for that component. Use signals to allow the component to interact with the outside without having code outside the component like this.